examples/siamese/create_db.py

```diff
@@ -0,0 +1,253 @@
+#!/usr/bin/env python
+# Copyright (c) 2015, NVIDIA CORPORATION.  All rights reserved.
+"""
+Functions for creating temporary LMDBs
+Used in test_views
+"""
+
+import os
+import sys
+import time
+import argparse
+from collections import defaultdict
+from cStringIO import StringIO
+import re
+import random
+
+import numpy as np
+import PIL.Image
+import lmdb
+
+if __name__ == '__main__':
+    dirname = os.path.dirname(os.path.realpath(__file__))
+    sys.path.insert(0, os.path.join(dirname,'..','..'))
+    from digits.config.load  import load_config
+    load_config()
+
+from digits import utils
+
+# Run load_config() first to set the path to Caffe
+import caffe.io
+import caffe_pb2
+
+IMAGE_SIZE  = 10
+TRAIN_IMAGE_COUNT = 1000
+VAL_IMAGE_COUNT = 1000
+
+
+def create_lmdbs(folder, file_list, image_width=None, image_height=None, image_count=None):
+    """
+    Creates LMDBs for generic inference
+    Returns the filename for a test image
+
+    Creates these files in "folder":
+        train_images/
+        train_labels/
+        val_images/
+        val_labels/
+        mean.binaryproto
+        test.png
+    """
+    if image_width is None:
+        image_width = IMAGE_SIZE
+    if image_height is None:
+        image_height = IMAGE_SIZE
+
+    if image_count is None:
+        train_image_count = TRAIN_IMAGE_COUNT
+    else:
+        train_image_count = image_count
+    val_image_count = VAL_IMAGE_COUNT
+
+    # read file list
+    images = []
+    f = open(file_list)
+    for line in f.readlines():
+        line = line.strip()
+        if not line:
+            continue
+
+        path = None
+        # might contain a numerical label at the end
+        match = re.match(r'(.*\S)\s+(\d+)$', line)
+        if match:
+            path = match.group(1)
+            ground_truth = int(match.group(2))
+            images.append([path,ground_truth])
+
+    print "Found %d image paths in image list" % len(images)
+
+    for phase, image_count in [
+            ('train', train_image_count),
+            ('val', val_image_count)]:
+
+        print "Will create %d pairs of %s images" % (image_count, phase)
+
+        # create DBs
+        image_db = lmdb.open(os.path.join(folder, '%s_images' % phase),
+                map_async=True,
+                max_dbs=0)
+        label_db = lmdb.open(os.path.join(folder, '%s_labels' % phase),
+                map_async=True,
+                max_dbs=0)
+
+        # add up all images to later create mean image
+        image_sum = None
+        shape = None
+
+        # save test images (one for each label)
+        testImageSameClass = None
+        testImageDifferentClass = None
```

Comment:
Can we create a few more test images? They're really hard to create without this script, and I find myself wanting to play with a few more examples.